tck/features/MatchAcceptance2.feature

```diff
@@ -0,0 +1,2023 @@
+      | b |
+    And no side effects
+
+  Scenario: Matching from null nodes should find nothing, filtered out
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ()-[:T]->()
+      """
+    When executing query:
+      """
+      OPTIONAL MATCH (a:Label)
+      WITH a
+      MATCH (a)-->(b)
+      RETURN b
+      """
+    Then the result should be:
+      | b |
+    And no side effects
+
+  Scenario: Optionally matching from null nodes should give null
+    Given an empty graph
+    When executing query:
+      """
+      OPTIONAL MATCH (a)
+      WITH a
+      OPTIONAL MATCH (a)-->(b)
+      RETURN b
+      """
+    Then the result should be:
+      | b    |
+      | null |
+    And no side effects
+
+  Scenario: OPTIONAL MATCH returning null
+    Given an empty graph
+    When executing query:
+      """
+      OPTIONAL MATCH (a)
+      RETURN a
+      """
+    Then the result should be:
+      | a    |
+      | null |
+    And no side effects
+
+  Scenario: Zero-length named path
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ()
+      """
+    When executing query:
+      """
+      MATCH p = (a)
+      RETURN p
+      """
+    Then the result should be:
+      | p    |
+      | <()> |
+    And no side effects
+
+  Scenario: Variable-length named path
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ()
+      """
+    When executing query:
+      """
+      MATCH p = ()-[*0..]->()
+      RETURN p
+      """
+    Then the result should be:
+      | p    |
+      | <()> |
+    And no side effects
+
+  Scenario: Matching with aggregation
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ({prop: 42})
+      """
+    When executing query:
+      """
+      MATCH (n)
+      RETURN n.prop AS n, count(n) AS count
+      """
+    Then the result should be:
+      | n  | count |
+      | 42 | 1     |
+    And no side effects
+
+  Scenario: Matching using already bound relationship
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ()-[:T1]->(),
+             ()-[:T2]->()
+      """
+    When executing query:
+      """
+      MATCH ()-[r1]->()
+      WITH r1 AS r2
+      MATCH ()-[r2]->()
+      RETURN r2 AS rel
+      """
+    Then the result should be:
+      | rel   |
+      | [:T1] |
+      | [:T2] |
+    And no side effects
+
+  Scenario: Matching using already bound relationship with aggregation
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ()-[:T1]->(),
+             ()-[:T2]->()
+      """
+    When executing query:
+      """
+      MATCH ()-[r1]->()
+      WITH r1 AS r2, count(*) AS c
+        ORDER BY c
+      MATCH ()-[r2]->()
+      RETURN r2 AS rel
+      """
+    Then the result should be:
+      | rel   |
+      | [:T1] |
+      | [:T2] |
+    And no side effects
+
+  Scenario: Matching using already bound relationship with aggregation and ORDER BY
```

Comment:
"Matching using already bound relationship with aggregation and ORDER BY" -> "Matching using a relationship that is already bound, in conjunction with aggregation and ORDER BY"